examples/camera-app/linux/src/camera-device.cpp

```diff
@@ -41,13 +42,59 @@ using namespace Camera;
 
 namespace {
 
+// Context structure to pass both CameraDevice and streamID to the callback
+struct AppSinkContext
+{
+    CameraDevice * device;
+    uint16_t streamID;
```

Comment:
Would recommend explicitly naming this as the videoStreamID for clarity. What about the audioStreamID?

Reply:
Acked, audioStream is not added at this time, the relevant change is not trivial, will do it in a separate PR